### DISCUSSION TITLE
Accessing redis cache in hooks and endpoints

### DISCUSSION THREAD

**Discussion by girishnuli** [directus/directus#16781] [Archive]
Is it possible to access redis cache as an injected service from within hooks and api endpoints (similar to how database is available as an injected service in the register function)?

Or is it better to keep custom cache usage separate from directus and access it directly by installing the node-redis package?

**Comment by paescuj** [comment]
Just out of curiosity: What's the use case for this?

**Reply by girishnuli** [reply]
We have pub-sub services that get triggered based on user actions. For example, a user can start some task (by calling the API) that involves multiple sub tasks. We initiate the process by publishing to a redis channel. Other micro-services subscribe to this channel to start their work.

We can always install node-redis separately and work with it, but since it is already included in directus, it would be helpful to expose it as a service in hooks and endpoints.

**Reply by NilsBaumgartner1994** [reply]
If implementing a PKCE, we would need to save some values in a storage, where multiple instances would have access to.

**Comment by jeffersonsolorzano** [comment]
Dis you find a way to access the cache in hook?

**Reply by girishnuli** [reply]
Nope. Not using this.
